src/bp-document/bp-document-functions.php

```diff
@@ -4238,12 +4238,12 @@ function bp_document_get_preview_url( $document_id, $attachment_id, $size = 'bb-
 				$file_path = $file_path . '/' . $file['file'];
 			}
 
-			if ( $file && ! empty( $file['file'] ) && ( ! empty( $file['path'] ) || ! file_exists( $file_path ) ) && 'pdf' !== $extension ) {
+			if ( $file && ! empty( $file['file'] ) && ( empty( $file['path'] ) || ! file_exists( $file_path ) ) && 'pdf' !== $extension ) {
 				// Regenerate attachment thumbnails.
 				bb_document_regenerate_attachment_thumbnails( $attachment_id );
 				$file      = image_get_intermediate_size( $attachment_id, $size );
 				$file_path = $file_path . '/' . $file['file'];
-			} elseif ( $file && ! empty( $file['file'] ) && ( ! empty( $file['path'] ) || ! file_exists( $file_path ) ) && 'pdf' === $extension ) {
+			} elseif ( $file && ! empty( $file['file'] ) && ( empty( $file['path'] ) || ! file_exists( $file_path ) ) && 'pdf' === $extension ) {
```

Comment:
@mahedihasannoman why both if conditions is same. Can you check once again.

Reply:
Hi @dipskakadiya , Both are not same. Difference is 'pdf' !== $extension & 'pdf' === $extension

Reply:
@mahedihasannoman I think in that case common condition add as a parent if and we should add inner if. For good code quality. just a structure change.

Reply:
Hi @dipskakadiya , Condition modified according to your suggestion.